File: crates/index/tests/fixtures/sample.go
package geometry

import (
	"fmt"
	"math"
)

// Shape is the common interface for geometric shapes.
type Shape interface {
	Area() float64
	Perimeter() float64
	Name() string
}

// Point represents a 2D coordinate.
type Point struct {
	X float64
	Y float64
}

// Circle is a round shape defined by a center and radius.
type Circle struct {
	Center Point
	Radius float64
}

// Rectangle is an axis-aligned rectangle.
type Rectangle struct {
	TopLeft     Point
	BottomRight Point
}

// Area returns the area of the circle.
func (c Circle) Area() float64 {
	return math.Pi * c.Radius * c.Radius
}

// Perimeter returns the circumference of the circle.
func (c Circle) Perimeter() float64 {
	return 2 * math.Pi * c.Radius
}

// Name returns the shape name.
func (c Circle) Name() string {
	return "circle"
}

// Area returns the area of the rectangle.
func (r Rectangle) Area() float64 {
	width := math.Abs(r.BottomRight.X - r.TopLeft.X)
	height := math.Abs(r.BottomRight.Y - r.TopLeft.Y)
	return width * height
}

// Perimeter returns the perimeter of the rectangle.
func (r Rectangle) Perimeter() float64 {
	width := math.Abs(r.BottomRight.X - r.TopLeft.X)
	height := math.Abs(r.BottomRight.Y - r.TopLeft.Y)
	return 2 * (width + height)
}

// Name returns the shape name.
func (r Rectangle) Name() string {
	return "rectangle"
}

// NewCircle creates a new circle with the given radius centered at the origin.
func NewCircle(radius float64) Circle {
	return Circle{Center: Point{}, Radius: radius}
}

// NewRectangle creates a new rectangle from corner coordinates.
func NewRectangle(x1, y1, x2, y2 float64) Rectangle {
	return Rectangle{
		TopLeft:     Point{X: x1, Y: y1},
		BottomRight: Point{X: x2, Y: y2},
	}
}

// describe returns a human-readable description of a shape.
func describe(s Shape) string {
	return fmt.Sprintf("%s: area=%.2f perimeter=%.2f", s.Name(), s.Area(), s.Perimeter())
}

File: crates/index/tests/fixtures/sample.rs
use std::collections::HashMap;
use std::fmt;

/// A 2D point with x and y coordinates.
pub struct Point {
    pub x: f64,
    pub y: f64,
}

impl Point {
    /// Creates a new Point.
    pub fn new(x: f64, y: f64) -> Self {
        Point { x, y }
    }

    /// Euclidean distance to another point.
    pub fn distance(&self, other: &Point) -> f64 {
        ((self.x - other.x).powi(2) + (self.y - other.y).powi(2)).sqrt()
    }

    /// Returns the origin point (0, 0).
    pub fn origin() -> Self {
        Point { x: 0.0, y: 0.0 }
    }
}

impl fmt::Display for Point {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "({}, {})", self.x, self.y)
    }
}

/// A shape with a name and color.
pub trait Shape {
    fn area(&self) -> f64;
    fn perimeter(&self) -> f64;
    fn name(&self) -> &str;
}

/// Named colors.
pub enum Color {
    Red,
    Green,
    Blue,
    Custom(u8, u8, u8),
}

impl Color {
    pub fn as_rgb(&self) -> (u8, u8, u8) {
        match self {
            Color::Red => (255, 0, 0),
            Color::Green => (0, 255, 0),
            Color::Blue => (0, 0, 255),
            Color::Custom(r, g, b) => (*r, *g, *b),
        }
    }
}

pub mod geometry {
    use std::f64::consts::PI;

    pub fn circle_area(radius: f64) -> f64 {
        PI * radius * radius
    }

    pub fn rectangle_area(width: f64, height: f64) -> f64 {
        width * height
    }
}

/// Counts word frequencies in a string.
pub fn word_count(text: &str) -> HashMap<&str, usize> {
    let mut counts = HashMap::new();
    for word in text.split_whitespace() {
        *counts.entry(word).or_insert(0) += 1;
    }
    counts
}
